ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -2,7 +2,7 @@
 from typing import List, Dict, Deque, TypeVar, Generic, Tuple, Any
```

Comment:
@ervin I made a lot of changes to this file. Basically, I am now processing the terminated agents first and then the ones that are ongoing. This allows for some simplification of the code (I think) but the logic is duplicated. Please review this part!

Reply:
@vincentpierre I tried to improve this in this PR: https://github.com/Unity-Technologies/ml-agents/pull/3728

Also I noticed the stats_reporter "Environment/Episode Length" reporting was gone so I added it back 😄